modules/bibcheck/lib/plugins/move_pattern_to_field.py

```diff
@@ -0,0 +1,98 @@
+## License, or (at your option) any later version.
+##
+## Invenio is distributed in the hope that it will be useful, but
+## WITHOUT ANY WARRANTY; without even the implied warranty of
+## MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+## General Public License for more details.
+##
+## You should have received a copy of the GNU General Public License
+## along with Invenio; if not, write to the Free Software Foundation, Inc.,
+## 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+
+"""
+Bibcheck plugin to move values of fields matching some pattern to a new field.
+
+The new value is taken from the 'value' named group of 'pattern' (with the
+'(?P<value>regexp)' construct), if it matches the value of source_field and put
+into new_field. A subfield_filter can be specified to further restrict the
+source field and additional_subfields can optionally be set to a list of
+constant subfield (code, value) pairs to add.
+
+If there is a match, the original field is deleted. By default, the new field
+is only created if it does not result in a duplicate value. This behavior can
+be disabled by setting allow_duplicates to True.
+
+Example rules:
+
+    [move_KEKSCAN]
+
+    check = move_pattern_to_field
+    check.source_field = "8564_u"
+    check.new_field = "035__a"
+    check.pattern = "^(https?://)?www-lib.kek.jp/cgi-bin/img_index\\?(?P<value>(\\d{2})?\\d{7})$"
+    check.subfield_filter = ["y", "KEKSCAN"]
+    check.additional_subfields = [["9", "KEKSCAN"]]
+    check.allow_duplicates = false
+    filter_collection = HEP
+    filter_pattern = 8564_u:'kek.jp/cgi-bin' - 035__9:KEKSCAN
+
+"""
+
+
+import re
+
+from invenio.bibrecord import record_add_field, record_get_field_values
+
+def check_record(record, source_field, new_field, pattern, 
+                 subfield_filter=(None, None), additional_subfields=[], 
+                 allow_duplicates=False):
+
+    assert len(source_field) == 6
+    assert len(new_field) == 6
+
+    new_code = new_field[5]
+    new_field = {
+        'tag': new_field[:3],
+        'ind1': new_field[3],
+        'ind2': new_field[4],
+    }
+    delcount = 0
+    regex = re.compile(pattern)
+    existing_values = set(record_get_field_values(record, code=new_code, **new_field))
+
+    for pos, val in record.iterfield(source_field,
+                                     subfield_filter=subfield_filter):
+        if val:
+            match = regex.match(val)
+
+            if match:
+                new_subfields = []
+                new_value = match.group('value')
+
+                if allow_duplicates or (
+                        new_value != '' and new_value not in existing_values):
+                    existing_values.add(new_value)
+                    new_subfields.extend(additional_subfields)
+                    new_subfields.append((new_code, new_value),)
+                    record_add_field(record, subfields=new_subfields,
```

Comment:
OK, I will change all `bibrecord` functions to `AmendableRecord` methods as you indicated.